routing/datasource.go

```diff
@@ -386,6 +390,10 @@ func processPredicates(cpm map[string]PredicateSpec, defs []*eskip.Predicate) ([
 			return nil, 0, fmt.Errorf("failed to create predicate %q: %w", spec.Name(), err)
 		}
 
+		if ws, ok := spec.(WeightedPredicateSpec); ok {
+			weight += ws.Weight()
```

Comment:
This is enabling predicates to have their own weight

Reply:
Yes